### ISSUE TITLE
array overflow in Region::time_step

### ISSUE THREAD

**Issue by pranjalv123** [shigh/pswr-charm#12]
In the last if statement we have 

```
    ++curr_chunk;
    curr_chunk_ind = 0;
    curr_ind = chunk_start[curr_chunk];
    solver->set_dt(dt_vals[curr_chunk]);
```

which overflows chunk_start and dt_vals, and makes load balancing crash

**Comment by shigh** [comment]
Are you sure Region::reset is getting called after each iteration? That would cause curr_chunk to go outside the valid range. 
Maybe it would be useful to use at() instead of []. In hindsight I should have done that anyways.

**Comment by shigh** [comment]
@pranjalv123 , how are you testing load balancing?
I did a pretty good job of messing up the pupping branch, if you do not have any changes left on it can you delete it and we can debug this on something branched from the current master.

**Comment by pranjalv123** [comment]
to test lb:
./charmrun +p4 ./swr 3 2 40 ++local +balancer GreedyCommLB +LBDebug 3

**Comment by shigh** [comment]
How is the load balancer being triggered? I can not find any ResumeFromSync or AtSync calls.

It looks like when it fails is not consistent. There are unsafe states bettween iterations that could cause this.

**Comment by pranjalv123** [comment]
fixed in commit a04b4b69613e172d953e832fda6103ba876b7032
